app/src/main/java/nerd/tuxmobil/fahrplan/congress/reporting/TraceDroidEmailSender.kt

```diff
@@ -0,0 +1,77 @@
+package nerd.tuxmobil.fahrplan.congress.reporting
+
+import android.app.Activity
+import org.ligi.tracedroid.TraceDroid
+import nerd.tuxmobil.fahrplan.congress.R
+import android.content.DialogInterface
+import android.content.Intent
+import android.content.ActivityNotFoundException
+import android.widget.Toast
+import androidx.appcompat.app.AlertDialog
+import de.cketti.mailto.EmailIntentBuilder
+import nerd.tuxmobil.fahrplan.congress.BuildConfig
+import org.ligi.tracedroid.collecting.TraceDroidMetaInfo
+import java.io.File
+
+// This class supports translation and configuration via XML files.
+// The original TraceDroidEmailSender class is available here:
+// https://github.com/ligi/tracedroid
+object TraceDroidEmailSender {
+    fun sendStackTraces(context: Activity) {
+        val stackTraceFile: Array<File>? = TraceDroid.getStackTraceFiles()
+
+        if (stackTraceFile == null || stackTraceFile.isEmpty()) {
+            return
+        }
+
+        val appName = context.getString(R.string.app_name)
+        val emailAddress = BuildConfig.TRACE_DROID_EMAIL_ADDRESS
+        val dialogTitle = context.getString(
+            R.string.trace_droid_dialog_title, appName
+        )
+        val dialogMessage = context.getString(
+            R.string.trace_droid_dialog_message, appName
+        )
+        val buttonTitleSend = context.getString(
+            R.string.trace_droid_button_title_send
+        )
+        val buttonTitleLater = context.getString(
+            R.string.trace_droid_button_title_later
+        )
+        val buttonTitleNo = context.getString(
+            R.string.trace_droid_button_title_no
+        )
+        val maximumStackTracesCount = context.resources.getInteger(
+            R.integer.config_trace_droid_maximum_stack_traces_count
+        )
+
+        AlertDialog.Builder(context)
+            .setTitle(dialogTitle)
+            .setMessage(dialogMessage)
+            .setPositiveButton(buttonTitleSend) { _: DialogInterface, _: Int ->
+                val emailIntent = getEmailIntent(context, emailAddress, maximumStackTracesCount)
+                try {
+                    context.startActivity(emailIntent)
+                    TraceDroid.deleteStacktraceFiles()
+                } catch (e: ActivityNotFoundException) {
+                    val message = context.getString(R.string.trace_droid_no_email_app)
+                    Toast.makeText(context, message, Toast.LENGTH_SHORT).show()
+                }
+            }
+            .setNegativeButton(buttonTitleNo) { _: DialogInterface, _: Int -> TraceDroid.deleteStacktraceFiles() }
+            .setNeutralButton(buttonTitleLater) { _: DialogInterface, _: Int -> }
+            .show()
+    }
+
+    private fun getEmailIntent(
+        activity: Activity,
+        recipient: String,
+        maximumStackTracesCount: Int
+    ): Intent {
+        return EmailIntentBuilder.from(activity)
```

Comment:
Convert to expression body, remove return type.

Reply:
Maybe it would be good to just remove that Method, as it is just used in this one place and does not seem to have any meaningful job :thinking: